### PR TITLE
nodePackages.prisma: 5.6.0 -> 5.12.1

### PR DESCRIPTION
**Pull request by aqrln**
## Description of changes

Update `prisma-engines` and `nodePackages.prisma` from 5.6.0 to 5.12.1.

Supersedes https://github.com/NixOS/nixpkgs/pull/273625

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by aqrln** [comment]
cc @ivan

File: pkgs/development/tools/database/prisma-engines/default.nix [hunk 1]
```diff
@@ -14,13 +14,13 @@
...
   pname = "prisma-engines";
-  version = "5.6.0";
+  version = "5.9.1";
 
   src = fetchFromGitHub {
     owner = "prisma";
     repo = "prisma-engines";
     rev = version;
-    sha256 = "sha256-zyF2NAOPNJe23tIuOHalsvnYytALKZq8QY1L8EWJMno=";
+    sha256 = "sha256-jvoe0JcgzwfjPHx1oCM5KxdoBcshzszZi6060O+2q8s=";
```
Comment (dotlambda) [inline pkgs/development/tools/database/prisma-engines/default.nix:23]:
```suggestion
    hash = "sha256-jvoe0JcgzwfjPHx1oCM5KxdoBcshzszZi6060O+2q8s=";
```

**Review by dotlambda — COMMENTED** [review]
Please repackage using `buildNpmPackage`. You can copy what I did in https://github.com/NixOS/nixpkgs/pull/285146#issuecomment-1930725547

**Comment by aqrln** [comment]
@dotlambda I tried it in the past but the problem is `buildNpmPackage` doesn't support pnpm and there's no tooling to build pnpm projects in nixpkgs. Ignoring the lockfile and replacing it with a new `package-lock.json` is not something I'm super comfortable with because I've seen test failures with even minor and patch version updates in the past plenty of times, but maybe the risk is relatively low for transitive dependencies. What would you suggest?

**Comment by dotlambda** [comment]
> What would you suggest?

The same as above.

**Comment by aqrln** [comment]
Sorry I didn't have time to do the refactoring, it is a bit non-trivial indeed. Prisma 5.13.0 will be out tomorrow, let's update it first because the version in `master` is fairly old at this point, and then I'll give it a shot again.

**Review by donovanglover — APPROVED** [review]
Built on x86_64-linux LGTM

**Comment by dotlambda** [comment]
Result of `nixpkgs-review pr 287590` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>documenso</li>
    <li>prisma-engines</li>
  </ul>
</details>

**Pull request merged**